### PR TITLE
frontend/fix/#3165/rideHistory_screen_bugfixes

### PR DESCRIPTION
**Pull request by Anuragini-2506**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Bugfixes for rideHistory Screen mentioned in this issue #3165 
Screenshot 2023-09-13 at 9 25 38 PM

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## How did you test it?

Tested locally

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

https://github.com/nammayatri/nammayatri/assets/83864083/6466908c-0231-4850-996e-bc0c65b85bf3

### CODE REVIEW

File: Frontend/ui-driver/src/Helpers/Utils.js [hunk 1]
```diff
@@ -675,19 +675,30 @@ function rideLabelConfig(){
...
 
+function getFormattedLanguage(language){
+  if (language == "EN_US") return "en-us";
+  else if (language == "HI_IN") return "hi-in";
+  else if (language == "KN_IN") return "kn-in";
+  else if (language == "TA_IN") return "ta-in";
+  else if (language == "BN_IN") return "bn-in";
+  else return "en-us";
+}
+
```
Comment (ashishsingh101) [inline Frontend/ui-driver/src/Helpers/Utils.js:701]: we can use same function from common only

File: Frontend/ui-driver/src/Screens/Types.purs [hunk 2]
```diff
@@ -1456,7 +1456,7 @@ type LeaderBoardWeek = {
   , endMonth :: String
 }
 
-data DateSelector = DaySelector LeaderBoardDay | WeekSelector LeaderBoardWeek
+data DateSelector = DaySelector DateObject | WeekSelector LeaderBoardWeek
```
Comment (ashishsingh101) [inline Frontend/ui-driver/src/Screens/Types.purs:1459]: define same for week also

File: Frontend/ui-driver/src/Screens/RideHistoryScreen/Controller.purs [hunk 3]
```diff
@@ -189,6 +191,8 @@ eval (Scroll value) state = do
 
 eval (DatePickerAC (DatePickerModel.OnDateSelect idx item)) state = do
   let newState = state{datePickerState{activeIndex = idx, selectedItem = item},rideList = [], prestoListArrayItems = []}
+  _ <- pure $ spy "Hello" newState
+  _ <- pure $ spy "jasdk" $ getFormattedDate state.datePickerState.selectedItem.utcDate
```
Comment (ashishsingh101) [inline Frontend/ui-driver/src/Screens/RideHistoryScreen/Controller.purs:195]: remove logs

File: Frontend/ui-driver/src/Screens/RideHistoryScreen/Controller.purs [hunk 4]
```diff
@@ -48,6 +48,8 @@ import Screens (ScreenName(..), getScreen)
 import Services.API (RidesInfo(..), Status(..))
 import Storage (KeyStore(..), getValueToLocalNativeStore, setValueToLocalNativeStore)
 import Styles.Colors as Color
+import Debug (spy)
+import Engineering.Helpers.Commons (flowRunner, getDateFromObj, getFormattedDate, getNewIDWithTag)
```
Comment (ashishsingh101) [inline Frontend/ui-driver/src/Screens/RideHistoryScreen/Controller.purs:52]: remove unused imports

**Review by ashishsingh101 — CHANGES_REQUESTED** [review]

File: Frontend/ui-driver/src/Screens/Types.purs (hunk 2 above)
Comment (Anuragini-2506) [reply Frontend/ui-driver/src/Screens/Types.purs:1459]: done

File: Frontend/ui-driver/src/Screens/RideHistoryScreen/Controller.purs (hunk 3 above)
Comment (Anuragini-2506) [reply Frontend/ui-driver/src/Screens/RideHistoryScreen/Controller.purs:195]: done

File: Frontend/ui-driver/src/Screens/RideHistoryScreen/Controller.purs (hunk 4 above)
Comment (Anuragini-2506) [reply Frontend/ui-driver/src/Screens/RideHistoryScreen/Controller.purs:52]: done

File: Frontend/ui-driver/src/Helpers/Utils.js (hunk 1 above)
Comment (Anuragini-2506) [reply Frontend/ui-driver/src/Helpers/Utils.js:701]: done

File: Frontend/ui-common/src/Engineering/Helpers/Commons.js [hunk 5]
```diff
@@ -366,15 +366,67 @@ export const getDateFromObj = function (obj){
   return  yyyy + '-' + mm + '-' + dd;
 }
 
+function getFormattedLanguage(language){
```
Comment (MercyQueen) [inline Frontend/ui-common/src/Engineering/Helpers/Commons.js:369]: @Anuragini-2506 can you please add it for Malayalam language too.

Comment (Anuragini-2506) [reply Frontend/ui-common/src/Engineering/Helpers/Commons.js:369]: done

**Review by ashishsingh101 — APPROVED** [review]

**Review by MercyQueen — APPROVED** [review]

**Review by kavya-shree-s — APPROVED** [review]

**Pull request merged**